cmd/admin-subnet-register.go

```diff
@@ -0,0 +1,181 @@
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cmd
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/probe"
+	"github.com/minio/pkg/console"
+	"github.com/tidwall/gjson"
+)
+
+var adminSubnetRegisterCmd = cli.Command{
+	Name:         "register",
+	Usage:        "register cluster with subnet",
+	OnUsageError: onUsageError,
+	Action:       mainAdminRegister,
+	Before:       setGlobalsFromContext,
+	Flags:        append(subnetCommonFlags, globalFlags...),
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} TARGET
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. Register the cluster at 'play' MinIO server.
+     {{.Prompt}} {{.HelpName}} play
+  2. Register the cluster at 'play' MinIO server, naming it as 'play-cluster' on subnet
+     {{.Prompt}} {{.HelpName}} play --name play-cluster
+  3. Register the cluster at 'play' MinIO server, using the proxy 192.168.1.3:3128
+     {{.Prompt}} {{.HelpName}} play --subnet-proxy 192.168.1.3:3128
+`,
+}
+
+// checkAdminRegisterSyntax - validate arguments passed by a user
+func checkAdminRegisterSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) == 0 || len(ctx.Args()) > 1 {
+		cli.ShowCommandHelpAndExit(ctx, "register", 1) // last argument is exit code
+	}
+}
+
+// ClusterRegistrationReq - JSON payload of the subnet api for cluster registration
+// Contains a registration token created by base64 encoding  of the registration info
+type ClusterRegistrationReq struct {
+	Token string `json:"token"`
+}
+
+// ClusterRegistrationInfo - Information stored in the cluster registration token
+type ClusterRegistrationInfo struct {
+	DeploymentID string      `json:"deployment_id"`
+	ClusterName  string      `json:"cluster_name"`
+	UsedCapacity uint64      `json:"used_capacity"`
+	Info         ClusterInfo `json:"info"`
+}
+
+// ClusterInfo - The "info" sub-node of the cluster registration information struct
+// Intended to be extensible i.e. more fields will be added as and when required
+type ClusterInfo struct {
+	MinioVersion    string `json:"minio_version"`
+	NoOfServerPools int    `json:"no_of_server_pools"`
+	NoOfServers     int    `json:"no_of_servers"`
+	NoOfDrives      int    `json:"no_of_drives"`
+	NoOfBuckets     uint64 `json:"no_of_buckets"`
+	NoOfObjects     uint64 `json:"no_of_objects"`
+}
+
+// SubnetLoginReq - JSON payload of the subnet login api
+type SubnetLoginReq struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+// SubnetMFAReq - JSON payload of the subnet mfa api
+type SubnetMFAReq struct {
+	Username string `json:"username"`
+	OTP      string `json:"otp"`
+	Token    string `json:"token"`
+}
+
+func mainAdminRegister(ctx *cli.Context) error {
+	checkAdminRegisterSyntax(ctx)
+
+	offlineMode := ctx.Bool(("offline"))
+	if !offlineMode && !subnetReachable() {
+		console.Fatalln("Subnet not reachable.")
```

Comment:
Should we give the `--offline` hint?

Reply:
`--offline` is anyway shown as part of the usage, and need not be mentioned here.